test_echo.py

```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+from echo_client import client
+
+def test_message():
+    message = u'This is the message to send to the server'
+    cmessage = client(message)
+    assert message == cmessage
+
+def test_unicode_message():
+    message = u'I like the café that is down the street'
+    cmessage = client(message)
+    assert message == cmessage
```

Comment:
Due to your implementation of converting unicode messages to bytes, it would be good to put a test in here that uses an encoded byte string with non-ascii characters in it.  That would demonstrate your problem, and offer you a chance to fix it in a way that is robust to both unicode and bytes input.